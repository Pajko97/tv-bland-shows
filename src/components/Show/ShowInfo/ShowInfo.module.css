.content_column {
    display:flex;
    flex-direction: column;
    justify-content: flex-start;
    flex-basis: grow;
    width: 100%;
    border:1px solid black;
    padding:2rem;
    margin-right: 2rem;
}

.column_item {
    display: flex;
    flex-direction: row;
    align-items: center;
    flex-wrap: wrap;
    justify-content: space-between;
    border-bottom: 1px solid black;
}
.column_item span {
    margin-right: 0.25rem;
}

.item_key {
    color:black;
}

.item_value {
    color: red;
}

.item_value span {
}


@media only screen and (max-width: 800px) {
    .content_column {
        width: auto;
        margin-right: 0;
    }
}