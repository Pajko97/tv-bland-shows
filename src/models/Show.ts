interface SingleShow {
    averageRuntime: number,
    dvdCountry: null
    ended: string,
    externals: Object
    genres: Array<any>,
    id: number,
    image: {
        medium : string,
        original : string
    },
    language: string
    name: string,
    network: {
        country : object,
        id: number,
        name : string
    },
    officialSite: null,
    premiered: string,
    rating: {
        average : any
    },
    runtime: number,
    schedule: {
        time : string,
        days: [string]
    },
    status: String,
    summary: string,
    type: string,
    updated: number,
    url: string,
    webChannel: null,
    weight: number,
    _links: Object
}   

export default SingleShow