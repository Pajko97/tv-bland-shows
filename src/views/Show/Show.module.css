.show_wrapper {
  padding: 10rem;
  font-family: "Red Hat Display";
}

.show_header {
  display: flex;
}
.show_header img {
  margin-right: 5rem;
  min-height: 100%;
  max-width: 25rem;
  max-height: auto;
}

.show_basic_info {
  max-width: 40rem;
}
.show_basic_info p {
  line-height: 150%;
}

.show_basic_info .rate {
  margin-left: 0.5rem;
}

.show_details {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

@media only screen and (max-width: 800px) {
  .show_wrapper {
    padding: 3rem;
  }
  .show_basic_info {
    margin-top: 2rem;
  }
  .show_header {
    display: flex;
    flex-direction: column;
  }
  .show_header img {
    margin-right: 0;
  }
  .show_details {
    flex-direction: column;
  }
}
