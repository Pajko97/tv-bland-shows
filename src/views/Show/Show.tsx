import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router';
import API from '../../api/shows-api';
import SingleShow from '../../models/Show'
import striptags from '../../helpers/striptags'
import Rating from 'react-rating'
import ActorList from "../../components/Show/ActorList/ActorList"
import style from './Show.module.css'
import ShowInfo from '../../components/Show/ShowInfo/ShowInfo';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCoffee, faStar, faStarHalfStroke } from '@fortawesome/free-solid-svg-icons'

type Props = Object

function Show(props: Props) {
  const [show, setShow] = useState<SingleShow>()

  let { id }: any = useParams();
  let showId = parseInt(id)

  useEffect(() => {
    let api = new API();
    api.getShow(showId)
      .then(res => setShow(res.data))
  }, [])

  const calcRating = (rating: number) =>
	rating > 5 ? Math.round((rating / 2) * 10) / 10 : rating

  return (
    <React.Fragment>
      {show && <div className={style.show_wrapper}>
        <div>
          <div className={style.show_header}>
            <img src={show.image.medium} alt="show cover"></img>
            <div className={style.show_basic_info}>
            <Rating
              emptySymbol={<FontAwesomeIcon icon={faStarHalfStroke} />}
              readonly
              initialRating={calcRating(show.rating.average)}
              fullSymbol={<FontAwesomeIcon icon={faStar}/>}

            />
            <span className={style.rate}>{show.rating.average ? `${show.rating.average}/5` : 'Not provided'}</span>
            <h1>{show.name}</h1>
            <p>{striptags(show.summary)}</p>
            </div>
          </div>

          <div className={style.show_details}>
            <ShowInfo show={show} />
            <ActorList showId={showId} />
          </div>
        </div>
      </div>}
    </React.Fragment>
  );
}

export default Show

