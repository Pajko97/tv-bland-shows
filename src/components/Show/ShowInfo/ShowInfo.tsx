import Show from "../../../models/Show";
import style from "./ShowInfo.module.css";


type ComponentProps = {
  show: Show;
};

export default function ShowInfo(props: ComponentProps) {
  let show_data = props.show;
  console.log(show_data)
  let genres = show_data?.genres;
  let days_showing = show_data.schedule.days;
  

  return (
    <div className={style.content_column}>
      <h2>Show Info</h2>
      <div className={style.column_item}>
        <p className={style.item_key}>Streaming on</p>
        <p className={style.item_value}>{show_data.network.name}</p>
      </div>

      <div className={style.column_item}>
        <p className={style.item_key}>Schedule</p>
        <p className={style.item_value}>
          {days_showing.map((day: string, id:number) => {
            return <span key={id}>{day}</span>;
          })}
        </p>
      </div>

      <div className={style.column_item}>
        <p className={style.item_key}>Status</p>
        <p className={style.item_value}>{show_data.status}</p>
      </div>

      <div className={style.column_item}>
        <p className={style.item_key}>Genres</p>
        <p className={style.item_value}>
          {genres
            ? genres.map((genre: string, i: number) => {
                return <span key={i}>{genre}</span>;
              })
            : "No genre"}
        </p>
      </div>
    </div>
  );
}
