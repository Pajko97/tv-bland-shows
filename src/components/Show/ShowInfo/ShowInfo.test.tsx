import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import API from '../../../api/shows-api';
import Show from '../../../models/Show';
import ShowInfo from './ShowInfo'

it('render app', async () => {
    let render, show: Show;
    let api = new API()
    await api.getShow(6)
    .then((res) => {
        res.data = show;
      render = renderer.create(
        <BrowserRouter>
              <ShowInfo show={show}/>
        </BrowserRouter>
      ).toJSON();
    })
    .catch((error) => console.error(error));

  expect(render).toMatchSnapshot();

})