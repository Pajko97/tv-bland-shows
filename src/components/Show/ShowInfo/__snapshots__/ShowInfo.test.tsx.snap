// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render app 1`] = `
<div
  className="content_column"
>
  <h2>
    Show Info
  </h2>
  <div
    className="column_item"
  >
    <p
      className="item_key"
    >
      Streaming on
    </p>
    <p
      className="item_value"
    >
      The CW
    </p>
  </div>
  <div
    className="column_item"
  >
    <p
      className="item_key"
    >
      Schedule
    </p>
    <p
      className="item_value"
    >
      <span>
        Wednesday
      </span>
    </p>
  </div>
  <div
    className="column_item"
  >
    <p
      className="item_key"
    >
      Status
    </p>
    <p
      className="item_value"
    >
      Ended
    </p>
  </div>
  <div
    className="column_item"
  >
    <p
      className="item_key"
    >
      Genres
    </p>
    <p
      className="item_value"
    >
      <span>
        Action
      </span>
      <span>
        Adventure
      </span>
      <span>
        Science-Fiction
      </span>
    </p>
  </div>
</div>
`;
